langchain/output_parsers/json.py

```diff
@@ -9,14 +9,21 @@
 
 def parse_json_markdown(json_string: str) -> dict:
     # Try to find JSON string within triple backticks
-    match = re.search(r"```(json)?(.*?)```", json_string, re.DOTALL)
+
+    # The amount of backticks at the end of the LLM response can vary,
+    # so we use a regex to match what we want even if we have fewer than 3 backticks
+    # As a tradeoff, we might fetch some backticks in the command below: match.group(2)
+    # We will get rid of them using rstrip() later
+    match = re.search(r"```(json)?(.*)(`{1,3})?", json_string, re.DOTALL)
```

Comment:
Any reason for the `?` after ````(`{1,3})````? As-is, it accept strings which contain zero ending backticks? Seems like we want at least one.